### PR TITLE
Ensure is_verified on /_matrix/client/r0/room_keys/keys is a boolean

### PR DESCRIPTION
**Pull request by anoadragon453**
Fixes: https://github.com/matrix-org/synapse/issues/6977

Also tells the user whether they're missing the `version` query parameter instead of 500'ing (this confused me for a bit while testing).

I'm not sure how [this sytest](https://github.com/matrix-org/sytest/blob/aee12f9414e5b36cba569e3ca0220e8e2f995329/tests/41end-to-end-keys/07-backup.pl#L225) didn't catch this initially though.

### CODE REVIEW

File: synapse/rest/client/v2_alpha/room_keys.py [hunk 1]
```diff
@@ -190,6 +190,11 @@ def __init__(self, hs):
         user_id = requester.user.to_string()
         version = parse_string(request, "version")
```
Comment (clokep) [inline synapse/rest/client/v2_alpha/room_keys.py]: I think you can provide the `required` parameter to `parse_string` to do the same thing, not 100% sure though.

Comment (anoadragon453) [reply synapse/rest/client/v2_alpha/room_keys.py]: You are correct, good shout!

**Comment by clokep** [comment]
@anoadragon453 Looks OK but unit tests are failing?

**Comment by anoadragon453** [comment]
It was being flaky. Not the test mind you, but the build machine.

**Pull request merged**